stylix/hm/default.nix

```diff
@@ -0,0 +1,66 @@
+{ palette-generator, base16 }:
+{ config, lib, ... }@mod-args:
+
+let
+  inherit (lib) types;
+  from-nixos = mod-args ? "osConfig";
+  system-theme = config.stylix.useSystemTheme;
+  autoload = import ../autoload.nix { inherit lib; } "hm";
+in {
+  imports = [
+    ../pixel.nix
+    ../target.nix
+    ./fonts.nix
+    (import ./palette.nix { inherit palette-generator base16; })
+  ] ++ autoload;
+
+  options.stylix = {
+    useSystemTheme = lib.mkOption {
```

Comment:
It would be convenient if the system theme was set as the default value in Home Manager, rather than needing this option. I can see why it's necessary, as passing down a value for `base16Scheme` would prevent a new scheme being generated if `image` is changed.

Perhaps we should combine the `image`, `palette`, `polarity` and `base16Scheme` options into a single option, and provide functions for the different ways to create a value for said option.

Something like `generateScheme { image, polarity }` would take an image and return an attribute set of `{ image, polarity, scheme }`, where the scheme is automatically generated. `loadScheme { image, path }` would return the same attribute set but instead load a premade scheme from the given file path.

The `palette` option was never particularly useful, so I think it can be removed. Otherwise it can be another argument to `generateScheme`.

If going ahead with this, I would also like to take the opportunity to make the returned `polarity` switch between `"light"` and `"dark"` based on the actual colours in the scheme. Epiphany switches websites between light and dark mode based on that value, but currently `polarity` is not specified when using a premade scheme.

Of course this would be a breaking change, so we'd need to add error messages to the removed options explaining what has changed.

Reply:
I pushed a first attempt to handle that. Now most options in home-manager use the system value as default **if** home-manager is used from NixOS **and** `stylix.homeManageIntegration.enable` is set in NixOS (which is the case by default).

I removed `palette` (I am planning to do set up a proper warning if it is used by someone). Now in home-manager `base16Scheme` is handled such that its value by default is its value in NixOS if the image in home-manager is the same as the one in NixOS, otherwise it is the newly generated config. Is this closer to what you were imagining ?

Reply:
I can now switch onto this branch without making any changes in my configuration, which is a good thing.

I had been meaning to clean up the `image`, `polarity` and `base16Scheme` options anyway, so I might have a go at some further changes when I have time. For now, this setup works well.